dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/attribute/TrackedEntityAttributeControllerTest.java

```diff
@@ -0,0 +1,73 @@
+ *
+ * Redistributions in binary form must reproduce the above copyright notice,
+ * this list of conditions and the following disclaimer in the documentation
+ * and/or other materials provided with the distribution.
+ * Neither the name of the HISP project nor the names of its contributors may
+ * be used to endorse or promote products derived from this software without
+ * specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ * ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR
+ * ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+ * (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+ * ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package org.hisp.dhis.webapi.controller.attribute;
+
+import static org.hisp.dhis.webapi.utils.WebClientUtils.assertStatus;
+
+import org.hisp.dhis.common.Objects;
+import org.hisp.dhis.schema.descriptors.TrackedEntityAttributeSchemaDescriptor;
+import org.hisp.dhis.textpattern.TextPattern;
+import org.hisp.dhis.textpattern.TextPatternParser;
+import org.hisp.dhis.trackedentity.TrackedEntityAttribute;
+import org.hisp.dhis.webapi.DhisControllerConvenienceTest;
+import org.hisp.dhis.webapi.utils.TestUtils;
+import org.junit.jupiter.api.Test;
+
+/**
+ * @author Luca Cambi
+ */
+class TrackedEntityAttributeControllerTest extends DhisControllerConvenienceTest
+{
+    @Test
+    void shouldGenerateRandomValuesOrgUnitCodeAndRandom()
+        throws Exception
+    {
+
+        TrackedEntityAttribute trackedEntityAttribute = createTrackedEntityAttribute( 'A' );
+        trackedEntityAttribute.setGenerated( true );
+
+        String pattern = "ORG_UNIT_CODE() + RANDOM(#######)";
+
+        TextPattern textPattern = TextPatternParser.parse( pattern );
+
+        textPattern.setOwnerObject( Objects.fromClass( trackedEntityAttribute.getClass() ) );
+        textPattern.setOwnerUid( trackedEntityAttribute.getUid() );
+
+        trackedEntityAttribute.setTextPattern( textPattern );
+        trackedEntityAttribute.setPattern( pattern );
+
+        String uid = assertStatus( org.springframework.http.HttpStatus.CREATED,
+            POST( TrackedEntityAttributeSchemaDescriptor.API_ENDPOINT,
+                new String( TestUtils.convertObjectToJsonBytes( trackedEntityAttribute ) ) ) );
+
+        assertStatus( org.springframework.http.HttpStatus.OK, GET(
+            TrackedEntityAttributeSchemaDescriptor.API_ENDPOINT + "/" + uid + "/generateAndReserve"
```

Comment:
just in case you did not know 😄  you can also use placeholders construct the URL https://github.com/dhis2/dhis2-core/blob/d654e335f20678b18edca3ad1c9787f455b5b82b/dhis-2/dhis-web-api-test/src/test/java/org/hisp/dhis/webapi/controller/AbstractCrudControllerTest.java#L129